rest/config_legacy.go

```diff
@@ -97,11 +100,43 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	// find a database's credentials for bootstrap (this isn't the first database config entry due to map iteration)
 	bsc := &BootstrapConfig{}
 	for _, dbConfig := range lc.Databases {
-		if dbConfig.Server == nil || *dbConfig.Server == "" {
+		server := dbConfig.Server
+		if server == nil || *server == "" {
 			continue
 		}
+
+		connSpec, err := gocbconnstr.Parse(*server)
+		if err != nil {
+			return nil, DbConfigMap{}, err
+		}
+
+		// support for username and password if not specified in username and password fields
+		splitServers := strings.Split(*server, ",")
+		// Only first URL is supported to have username and pass or else gocbconnstr.Parse does not work correctly
```

Comment:
This seems unexpected to me, but might just be undocumented behaviour of how it (or SG) used to work.

Would be worth double checking if SG 2.8 supports running with a username/password in a host that isn't first.

Did you mean `url.Parse` fails for comma-separated hosts, rather than `gocbconnstr.Parse`?

Reply:
`gocbconnstr.Parse` does not work as intended when the username and password is not in the first server address.

After testing with SGW 2.8, the username and password only works if it is in the first server URL and then this username and password is used for the fallback URLs as well.